### PR TITLE
Topic api 3

### PR DESCRIPTION
**Pull request by hallnsh**

### CODE REVIEW

File: README-SAVED.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+git branch -D <local branch>
+```
+
+## Husky
+
+To ensure we are not commiting broken code this project makes use of git hooks. Git hooks are scripts triggered during certain events in the git lifecycle. Husky is a popular package which allows us to set up and maintain these scripts. This project makes use a _pre-commit hook_. When we attempt to commit our work, the script defined in the `pre-commit` file will run. If any of our tests fail than the commit will be aborted.
+
+The [Husky documentation](https://typicode.github.io/husky/#/) explains how to configure Husky for your own project as well as creating your own custom hooks.\_
+
+###  typos
```
Comment (daviidbartlett) [inline README-SAVED.md]: this readme should have been deleted 👀

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+const request = require('supertest');
+const app = require('../app');
+const seed = require('../db/seeds/seed');
+const data = require('../db/data/test-data');
+const db = require('../db/connection');
+
+afterAll(() => {
+    if (db.end) db.end();
+  });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:9]: why are you asking if (db.end)?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        });
+        test(`status:200, responds with array of topic objects: properites 
+                slug: string, description: string`, () => {
+            return request(app)
+              .get('/api/topics')
+              .expect(200)
+              .then(({ body }) => {
+                const { topics } = body;
+                expect(topics).toBeInstanceOf(Array);
+                topics.forEach((topic) => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:31]: you already have this assertion in the above test

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+              });
+        });
+        test('status:200, responds with an array of length 3 topic objects', () => {
+            return request(app)
+              .get('/api/topics')
+              .expect(200)
+              .then(({ body }) => {
+                const { topics } = body;
+                expect(topics).toBeInstanceOf(Array);
+                console.log(topics[0]);
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: remove console.log

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                        });
+              });
+        });
+        test('status:200, responds with an array of length 3 topic objects', () => {
+            return request(app)
+              .get('/api/topics')
+              .expect(200)
+              .then(({ body }) => {
+                const { topics } = body;
+                expect(topics).toBeInstanceOf(Array);
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: repeated assertion

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                console.log(topics[0]);
+                expect(topics).toHaveLength(3);
+              });
+        });
+        test('status:404, responds with path not found message for incorrect path', () => {
+            return request(app)
+              .get('/not-an-endpoint')
+              .expect(404)
+              .then(({ body: {msg} }) => {
+                  console.log('ERROR MESSAGE -----',{body: {msg}});
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: remove console.log

File: __tests__/app.test.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                expect(msg).toBe('path not found');
+              });
+        });
+        // test('status:500, responds with server error message for any other problem with server', () => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: not a test you can do

File: controllers/topics.controllers.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const {fetchTopics} = require('../models/topics.models');
+
+exports.getTopics = (req, res, next) => {
+    fetchTopics()
+      .then((topics) => {
+      /**
+       * implement the fetchTopics model function and complete the controller
+       * by responding with the topics to pass the first set of tests
+       */
+        console.log ('In getTopics');
```
Comment (daviidbartlett) [inline controllers/topics.controllers.js]: remove

File: controllers/topics.controllers.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        res.status(200)
+        res.send({topics});
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (daviidbartlett) [inline controllers/topics.controllers.js]: remove

File: models/topics.models.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+const db = require('../db/connection');
+
+exports.fetchTopics = () => {
+    return db
+    .query('SELECT * FROM topics;')
+    .then (({rows}) => {
+        console.log('Inside fetchTopics');
```
Comment (daviidbartlett) [inline models/topics.models.js]: remove

**Review by daviidbartlett — COMMENTED** [review]
generally good! a few tidy ups

File: README-SAVED.md (hunk 1 above)
Comment (hallnsh) [reply README-SAVED.md]: This is not the README.md, it is a saved copy but I will kill it!

File: __tests__/app.test.js (hunk 2 above)
Comment (hallnsh) [reply __tests__/app.test.js:9]: Oh I will kill this! thanks

File: __tests__/app.test.js (hunk 7 above)
Comment (hallnsh) [reply __tests__/app.test.js]: This led me to NCHELP and also led me a merry dance through docs!

File: models/topics.models.js (hunk 10 above)
Comment (hallnsh) [reply models/topics.models.js]: console.log removed

File: controllers/topics.controllers.js (hunk 9 above)
Comment (hallnsh) [reply controllers/topics.controllers.js]: console.log removed

File: controllers/topics.controllers.js (hunk 8 above)
Comment (hallnsh) [reply controllers/topics.controllers.js]: removed

File: __tests__/app.test.js (hunk 6 above)
Comment (hallnsh) [reply __tests__/app.test.js]: removed console.log

File: __tests__/app.test.js (hunk 7 above)
Comment (hallnsh) [reply __tests__/app.test.js]: removed the commented block that was looking to test for a 500

File: __tests__/app.test.js (hunk 6 above)
Comment (hallnsh) [reply __tests__/app.test.js]: removed console.log

File: __tests__/app.test.js (hunk 5 above)
Comment (hallnsh) [reply __tests__/app.test.js]: removed the repeated assertion, asserting instance of array

File: __tests__/app.test.js (hunk 4 above)
Comment (hallnsh) [reply __tests__/app.test.js]: removed console.log

File: __tests__/app.test.js (hunk 3 above)
Comment (hallnsh) [reply __tests__/app.test.js:31]: fixed this, I didn't notice the repeat

File: __tests__/app.test.js (hunk 2 above)
Comment (hallnsh) [reply __tests__/app.test.js:9]: fixed this!

**Review by scar1377 — APPROVED** [review]
README--SAVED.md is still on Github. Apart from that, OK to merge

**Pull request merged**